// react-github-activity/lib/head/feedHeader.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; desc = parent = undefined; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ghIcon = require('./ghIcon');

var _ghIcon2 = _interopRequireDefault(_ghIcon);

var _profileDetails = require('./profileDetails');

var _profileDetails2 = _interopRequireDefault(_profileDetails);

var _avatar = require('./avatar');

var _avatar2 = _interopRequireDefault(_avatar);

var FeedHeader = (function (_Component) {
  _inherits(FeedHeader, _Component);

  function FeedHeader() {
    _classCallCheck(this, FeedHeader);

    _get(Object.getPrototypeOf(FeedHeader.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(FeedHeader, [{
    key: 'render',
    value: function render() {
      return _react2['default'].createElement(
        'div',
        { style: this.props.styles.githubHeaderOuter, className: 'github-header-outer' },
        _react2['default'].createElement(
          'div',
          { style: this.props.styles.githubHeader, className: 'github-header' },
          _react2['default'].createElement(_ghIcon2['default'], { styles: this.props.styles }),
          _react2['default'].createElement(_profileDetails2['default'], {
            fullName: this.props.fullName,
            userName: this.props.userName,
            profileUrl: this.props.profileUrl,
            avatarUrl: this.props.avatarUrl,
            styles: this.props.styles
          }),
          _react2['default'].createElement(_avatar2['default'], {
            avatarUrl: this.props.avatarUrl,
            styles: this.props.styles
          })
        )
      );
    }
  }]);

  return FeedHeader;
})(_react.Component);

exports['default'] = FeedHeader;
module.exports = exports['default'];
