// react-github-activity/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; desc = parent = undefined; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _headFeedHeader = require('./head/feedHeader');

var _headFeedHeader2 = _interopRequireDefault(_headFeedHeader);

var _bodyEventList = require('./body/eventList');

var _bodyEventList2 = _interopRequireDefault(_bodyEventList);

var _footerFooter = require('./footer/footer');

var _footerFooter2 = _interopRequireDefault(_footerFooter);

var _styles = require('./styles');

var GithubFeed = (function (_Component) {
  _inherits(GithubFeed, _Component);

  function GithubFeed() {
    _classCallCheck(this, GithubFeed);

    _get(Object.getPrototypeOf(GithubFeed.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(GithubFeed, [{
    key: 'render',
    value: function render() {
      var _props = this.props;
      var fullName = _props.fullName;
      var userName = _props.userName;
      var avatarUrl = _props.avatarUrl;
      var profileUrl = _props.profileUrl;
      var events = _props.events;

      return _react2['default'].createElement(
        'div',
        { style: _styles.styles.githubFeed },
        _react2['default'].createElement(_headFeedHeader2['default'], {
          fullName: fullName,
          userName: userName,
          avatarUrl: avatarUrl,
          profileUrl: profileUrl,
          styles: _styles.styles
        }),
        _react2['default'].createElement(_bodyEventList2['default'], {
          events: events,
          styles: _styles.styles
        }),
        _react2['default'].createElement(_footerFooter2['default'], {
          styles: _styles.styles
        })
      );
    }
  }]);

  return GithubFeed;
})(_react.Component);

exports['default'] = GithubFeed;
module.exports = exports['default'];
